app/assets/javascripts/3dbio_viewer/src/webapp/components/annotations-tool/AnnotationsTool.tsx

```diff
@@ -1,71 +1,87 @@
-import React, { useState, useRef } from "react";
+import React, { useCallback, useState, useRef } from "react";
 import _ from "lodash";
 import { Dialog, DialogContent, DialogTitle, IconButton, Switch } from "@material-ui/core";
 import { Close } from "@material-ui/icons";
 import i18n from "../../utils/i18n";
+import { useBooleanState } from "../../hooks/use-boolean";
 import { Dropzone, DropzoneRef } from "../dropzone/Dropzone";
 import { ProtvistaAction } from "../protvista/Protvista.helpers";
 import "./AnnotationsTool.css";
 
-export interface ModelUploadProps {
-    title: string;
+export interface AnnotationsToolProps {
     onClose(): void;
     action: ProtvistaAction;
 }
 
-export const AnnotationsTool: React.FC<ModelUploadProps> = React.memo(props => {
-    const { title, onClose, action } = props;
+interface AnnotationForm {
+    trackName: string;
+    type: string;
+    description: string;
+    color: string;
+    index: string;
```

Comment:
`index` has 2 only to valid values, so we may narrow the type with a little bit of typescript machinery. Check my commit and let me know if it makes sense to you.